### PR TITLE
Don't allow envoy sidecar privilege escalation

### LINKED ISSUE #4850: Disallow privilege escalation on envoy sidecar
**Bug description**:
The only container in all of OSM that requires anything resembling admin privileges is the init container. When we inject the Envoy sidecar, we need to set the [allowPrivilegeEscalation](https://kubernetes.io/docs/concepts/security/pod-security-policy/#privilege-escalation) field on the pod to false.

**Affected area (please mark with X where applicable)**:
- Envoy Control Plane        [X]

### PR DESCRIPTION
**Pull request by keithmattix**
Signed-off-by: Keith Mattix II <keithmattix2@gmail.com>

**Description**:
Fixes #4850

**Testing done**:
Added unit test for envoy sidecar OS specific components

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Security                   | [X] |
| Sidecar Injection          | [X] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? no
    -   Did you notify the maintainers and provide attribution? N/A

2. Is this a breaking change? no

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? no

### CODE REVIEW

**Review by shashankram — APPROVED** [review]

File: pkg/injector/envoy_container_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/golang/mock/gomock"
+	tassert "github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+func Test_getPlatformSpecificSpecComponents(t *testing.T) {
```
Comment (shalier) [inline pkg/injector/envoy_container_test.go:15]:
```suggestion
func TestGetPlatformSpecificSpecComponents(t *testing.T) {
```

**Review by trstringer — APPROVED** [review]
Approved, but agree with @shalier's requested change.

File: pkg/injector/envoy_container_test.go (hunk 1 above)
Comment (keithmattix) [reply pkg/injector/envoy_container_test.go]: Good catch! _shakes fist at VSCode's test generation_

Comment (keithmattix) [reply pkg/injector/envoy_container_test.go]: _sigh_ accepting PR suggestions breaks DCO; of course

**Pull request merged**